include/swift/Basic/Located.h

```diff
@@ -0,0 +1,39 @@
+//===--- Located.h - Source Location and Associated Value ----------*- C++ -*-===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2019 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+//
+// This file forward declares and imports various common LLVM datatypes that
+// swift wants to use unqualified.
+//
+//===----------------------------------------------------------------------===//
+
+
+#ifndef SWIFT_BASIC_LOCATED_H
+#define SWIFT_BASIC_LOCATED_H
+#include "swift/Basic/SourceLoc.h"
+
+namespace swift {
+
+template<typename T>
+struct Located {
+
+  T item;
+
+  SourceLoc loc;
+
+  template<typename U>
+  friend bool operator ==(const Located<U> lhs, const Located<U> rhs) {
```

Comment:
@varungandhi-apple There are a lot of places, where we use == with pair, should we remove this operator overloading and try to rewrite those places, without it ?

Reply:
It's ok to have this overload. 👍 I did not realize that we were using `==` for `std::pair` but this makes sense.

Reply:
Oh wait, could you mark the `lhs` and `rhs` as being passed by reference instead of by value? Don't want to accidentally copy the item in case it is a chonky string.

Reply:
Sure. Fixed.